src/lib.rs

```diff
@@ -40,81 +60,162 @@ impl<T: AsRef<[u8]>> Display for HexFmt<T> {
 
 impl<T: AsRef<[u8]>> LowerHex for HexFmt<T> {
     fn fmt(&self, f: &mut Formatter) -> Result {
+        Lowercase::fmt(self.0.as_ref(), f)
+    }
+}
+
+impl<T: AsRef<[u8]>> UpperHex for HexFmt<T> {
+    fn fmt(&self, f: &mut Formatter) -> Result {
+        Uppercase::fmt(self.0.as_ref(), f)
+    }
+}
+
+/// Wrapper for a list of byte arrays, whose `Debug`, `Display` and `LowerHex` implementations
+/// output shortened hexadecimal strings.
+pub struct HexList<T>(pub T);
+
+impl<T: Clone + IntoIterator> Debug for HexList<T>
```

Comment:
Would it not be possible to implement this without the `Clone` and `IntoIterator` requirements?

Reply:
I'd like to have it as general as possible: You should be able to pass in a `Vec<Bytes>`, `&[Bytes]`, `&BTreeSet<Bytes>` or `btree_set::Iter<&Bytes>`.

If the trait bound is just `Iterator`, the first three wouldn't be possible, so it needs to be `IntoIterator`. And `Clone` is needed because `into_iter` consumes its receiver.